scripts/pylint_custom_plugin/pylint_guidelines_checker.py

```diff
@@ -1705,6 +1705,131 @@ def visit_functiondef(self, node):
     # this line makes it work for async functions
     visit_asyncfunctiondef = visit_functiondef
 
+class CheckEnum(BaseChecker):
+    __implements__ = IAstroidChecker
+
+    name = "check-enum"
+    priority = -1
+    msgs = {
+        "C4746": (
+            "The enum must use uppercase naming. "
+            "https://azure.github.io/azure-sdk/python_design.html#enumerations",
+            "enum-must-be-uppercase",
+            "Capitalize enum name.",
+        ),
+        "C4747":(
+            "The enum must inherit from CaseInsensitiveEnumMeta. "
+            "https://azure.github.io/azure-sdk/python_implementation.html#extensible-enumerations",
+            "enum-must-inherit-case-insensitive-enum-meta",
+            "Inherit CaseInsensitiveEnumMeta.",
+        ),
+    }
+    options = (
+        (
+            "ignore-enum-must-be-uppercase",
+            {
+                "default": False,
+                "type": "yn",
+                "metavar": "<y_or_n>",
+                "help": "Allow an enum to not be capitalized.",
+            },
+        ),
+        (
+            "ignore-enum-must-inherit-case-insensitive-enum-meta",
+            {
+                "default": False,
+                "type": "yn",
+                "metavar": "<y_or_n>",
+                "help": "Allow an enum to not inherit CaseInsensitiveEnumMeta.",
+            },
+        ),
+    )
+
+    def __init__(self, linter=None):
+        super(CheckEnum, self).__init__(linter)
+
+    def visit_classdef(self, node):
+        """Visits every enum class.
+
+        :param node: ast.ClassDef
+        :return: None
+        """
+        try:
+
+            inherits_metaclass, enum_class = self._check_is_enum_class(node)
+            
+            # If it has a metaclass, and is an enum class, check the capitalization
+            if inherits_metaclass and enum_class:
+                self._enum_uppercase(node)   
+            # Else if it does not have a metaclass, but it is an enum class
+            # Check both capitalization and throw pylint error for metaclass
+            elif enum_class:
+                self._does_not_inherit_case_insensitive(node)
+                self._enum_uppercase(node)   
+
+        except Exception:
+            logger.debug("Pylint custom checker failed to check enum.")
+            pass
+
+    def _check_is_enum_class(self,node):
+        """Checks if a class is an enum class.
+        Returns booleans for the two enum guidelines.
+
+        :param node: ast.ClassDef
+        :return: inherits_metaclass, enum_class
+        :rtype: bool, bool
+            
+        """
+
+        inherits_metaclass = False
+        enum_class = False
+
+        if node._metaclass:
+            if node._metaclass.name == "CaseInsensitiveEnumMeta":
+                inherits_metclass = True
```

Comment:
Nit: this var is not the same as the one declared on line1784 and the one returned on 1791. 

Same comment for line 1794 below. 

I'm curious if the tests are passing currently given that `inherits_metaclass` should be returning False in all cases then.

Reply:
Ah, sorry! I am going to change to the direct boolean to fix this.